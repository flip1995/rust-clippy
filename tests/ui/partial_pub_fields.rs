#![allow(unused)]
#![warn(clippy::partial_pub_fields)]

fn main() {
    // test code goes here

    use std::collections::HashMap;

    #[derive(Default)]
    pub struct FileSet {
        files: HashMap<String, u32>,
        pub paths: HashMap<u32, String>,
    }

    pub struct Color {
        pub r: u8,
        pub g: u8,
        b: u8,
    }
}
