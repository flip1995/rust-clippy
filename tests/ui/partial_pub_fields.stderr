error: mixed usage of pub and non-pub fields
  --> $DIR/partial_pub_fields.rs:12:9
   |
LL |         pub paths: HashMap<u32, String>,
   |         ^^^
   |
   = help: consider using private field here
   = note: `-D clippy::partial-pub-fields` implied by `-D warnings`

error: mixed usage of pub and non-pub fields
  --> $DIR/partial_pub_fields.rs:18:9
   |
LL |         b: u8,
   |         ^
   |
   = help: consider using public field here

error: aborting due to 2 previous errors

